packages/utils-tests-helpers/src/hint-runner.ts

```diff
@@ -79,6 +79,57 @@ const findPosition = (source: string, position: MatchProblemLocation): ProblemLo
     return result;
 };
 
+/**
+ * Compares two location to see if they have the same start/end values
+ */
+const comparePositions = (position1: ProblemLocation, position2: ProblemLocation): boolean => {
+    if (position1.line !== position2.line) {
+        return false;
+    }
+    if (position1.column !== position2.column) {
+        return false;
+    }
+    if (position1.endLine !== position2.endLine) {
+        return false;
+    }
+    if (position1.endColumn !== position2.endColumn) {
+        return false;
+    }
+
+    return true;
+};
+
+/**
+ * Translates line and column start and end values to offset values.
+ */
+const positionToOffset = (position: ProblemLocation, document: string): number[] => {
+    let startOffset = -1;
+    let endOffset = -1;
+
+    const {column, endColumn, endLine, line} = position;
+
+    if (!endLine || !endColumn) {
+        return [-1, -1];
+    }
+
+    const regex = /(\r\n|\r|\n)/gm;
+
+    for (let i = 0; i < endLine; i++) {
+        // We want to add 1 to the resulting index to account for the newline char. If the resulting index is undefined, we set curLineOffset to 0.
+        const curLineOffset = (regex.exec(document)?.index || -1) + 1;
```

Comment:
Per your comment, I suspect this only works with `\n` or `\r`, but not `\r\n`. Probably better to use `lastIndex` instead which automatically places you at the start of the next match so you don't have to manually account for which type of newline was used.

```suggestion
        const curLineOffset = regex.exec(document)?.index || -1;
```

Might also be worth exporting this function and writing a couple of basic unit tests for it to ensure we handle each newline case correctly.